proxy/src/registry/transaction.rs

```diff
@@ -127,6 +202,60 @@ where
                 .expect("unable to get 'transactions' bucket"),
         }
     }
+
+    /// Starts up the machinery to check for latest block information and advance the state of
+    /// cached transactions.
+    ///
+    /// # Errors
+    ///
+    /// Will return `Err` if a protocol error occurs or access to [`kv::Store`] fails.
+    pub async fn run(&self) -> Result<(), error::Error> {
+        let mut interval = tokio::time::interval(Duration::from_secs(10));
+        loop {
+            interval.tick().await;
+
+            self.advance(self.client.latest_height().await?)?;
+        }
+    }
+
+    /// Updates cached transactions progress given the latest height.
+    fn advance(&self, latest_height: protocol::BlockNumber) -> Result<(), error::Error> {
+        let mut txs = self.list_transactions(vec![])?;
+
+        for tx in &mut txs {
+            match tx.state {
+                State::Confirmed {
+                    block, timestamp, ..
+                } => {
+                    let target = block
+                        .checked_add(SETTLEMENT_OFFSET)
+                        .unwrap_or(SETTLEMENT_OFFSET);
```

Comment:
If `SETTLEMENT_OFFSET == MIN_CONFIRMATIONS - 1`, then this is correct. Otherwise it's off by one.